types/signed_header.go

```diff
@@ -9,3 +13,24 @@ func (sH *SignedHeader) New() header.Header {
 func (sH *SignedHeader) IsZero() bool {
 	return sH == nil
 }
+
+func (sH *SignedHeader) VerifyAdjacent(untrst header.Header) error {
+	// Explicit type checks are required due to embedded Header which also does the explicit type check
+	untrstH, ok := untrst.(*SignedHeader)
+	if !ok {
+		return &header.VerifyError{
+			Reason: fmt.Errorf("%T is not of type %T", untrst, sH),
+		}
+	}
+	return sH.Header.VerifyAdjacent(&untrstH.Header)
+}
+
+func (sH *SignedHeader) VerifyNonAdjacent(untrst header.Header) error {
+	untrstH, ok := untrst.(*SignedHeader)
```

Comment:
Does the same comment as before `Explicit type checks` apply here?

Reply:
yes